addons/website/tests/test_lang_url.py

```diff
@@ -118,32 +120,32 @@ def test_01_controller_redirect(self):
             the controller explicitly specifies a trailing slash in the route).
         """
 
-        def assertUrlRedirect(url, expected_url, msg="", code=301):
-            if expected_url.startswith('/'):
-                expected_url = "http://%s:%s%s" % (HOST, odoo.tools.config['http_port'], expected_url)
+        def assertUrlRedirect(url, expected_url, msg="", expected_query="", code=301):
             if not msg:
                 msg = 'Url <%s> differ from <%s>.' % (url, expected_url)
 
             r = self.url_open(url, head=True)
             self.assertEqual(r.status_code, code)
-            self.assertEqual(r.headers.get('Location'), expected_url, msg)
+            parsed_url = urlparse(r.headers.get('Location', ''))
+            self.assertEqual(parsed_url.query, expected_query)
+            self.assertEqual(parsed_url.path, expected_url, msg)
```

Comment:
So we don't have to change the function's signature and still parse both path and query
```py
url1 = urlparse(url)
url2 = urlparse(expected_url)
assertEqual(url1.path, url2.path, msg)
assertEqual(url1.query, url2.query, msg)

Reply:
Should restore the original code in the functions that use this `assertUrlRedirect` too

Reply:
```py
        def assertUrlRedirect(url, expected_url, msg="", code=301):
            if not msg:
                msg = 'Url <%s> differ from <%s>.' % (url, expected_url)

            r = self.url_open(url, head=True)
            self.assertEqual(r.status_code, code)

            parsed_expected_url = urlparse(expected_url)
            parsed_location = urlparse(r.headers.get('Location', ''))

            self.assertEqual(parsed_location.path, parsed_expected_url.path, msg)
            self.assertEqual(parsed_location.query, parsed_expected_url.query, msg)
```